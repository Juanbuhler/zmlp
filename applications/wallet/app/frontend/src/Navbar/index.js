import PropTypes from 'prop-types'

import userShape from '../User/shape'

import { colors, spacing, constants, zIndex } from '../Styles'

import LogoSvg from '../Icons/logo.svg'

import ProjectSwitcher from '../ProjectSwitcher'
import UserMenu from '../UserMenu'

import HamburgerSvg from './hamburger.svg'

const LOGO_WIDTH = 110

const Navbar = ({ user, isSidebarOpen, setSidebarOpen, logout }) => {
  return (
    <div
      css={{
        position: 'fixed',
        top: 0,
        left: 0,
        right: 0,
        height: constants.navbar.height,
        display: 'flex',
        justifyContent: 'space-between',
        alignItems: 'center',
        backgroundColor: colors.grey1,
        boxShadow: constants.boxShadows.navBar,
        zIndex: zIndex.layout.navbar,
        paddingLeft: spacing.normal,
        paddingRight: spacing.normal,
      }}>
      <div css={{ display: 'flex', alignItems: 'stretch' }}>
        <button
          aria-label="Open Sidebar Menu"
          type="button"
          onClick={() => setSidebarOpen(!isSidebarOpen)}
          css={{
            border: 0,
            backgroundColor: 'inherit',
            color: colors.structure.steel,
            display: 'flex',
            justifyContent: 'center',
            alignItems: 'center',
            padding: spacing.base,
            margin: 0,
            marginLeft: -spacing.base,
            cursor: 'pointer',
          }}>
          <HamburgerSvg width={20} aria-hidden />
        </button>

        <div css={{ paddingLeft: spacing.base, paddingRight: spacing.base }}>
          <LogoSvg width={LOGO_WIDTH} />
        </div>

        <ProjectSwitcher />
      </div>

      <UserMenu user={user} logout={logout} />
    </div>
  )
}

Navbar.propTypes = {
  user: PropTypes.shape(userShape).isRequired,
  isSidebarOpen: PropTypes.bool.isRequired,
  setSidebarOpen: PropTypes.func.isRequired,
  logout: PropTypes.func.isRequired,
}

export default Navbar
