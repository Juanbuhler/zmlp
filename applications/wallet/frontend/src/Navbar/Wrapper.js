import PropTypes from 'prop-types'

import { colors, spacing, constants, zIndex } from '../Styles'

const NavbarWrapper = ({ children, style }) => (
  <div
    css={{
      position: 'fixed',
      top: 0,
      left: 0,
      right: 0,
      height: constants.navbar.height,
      display: 'flex',
      justifyContent: 'space-between',
      alignItems: 'center',
      backgroundColor: colors.structure.mattGrey,
      boxShadow: constants.boxShadows.navBar,
      zIndex: zIndex.layout.navbar,
      paddingLeft: spacing.normal,
      paddingRight: spacing.normal,
      ...style,
    }}>
    {children}
  </div>
)

NavbarWrapper.defaultProps = {
  style: {},
}

NavbarWrapper.propTypes = {
  children: PropTypes.node.isRequired,
  style: PropTypes.shape({ name: PropTypes.string, styles: PropTypes.string }),
}

export default NavbarWrapper
