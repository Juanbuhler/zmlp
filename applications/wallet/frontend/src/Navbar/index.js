import PropTypes from 'prop-types'
import Link from 'next/link'

import { colors, spacing, constants, zIndex } from '../Styles'

import LogoSvg from '../Icons/logo.svg'
import HamburgerSvg from '../Icons/hamburger.svg'

import ProjectSwitcher from '../ProjectSwitcher'

const LOGO_HEIGHT = 18

const Navbar = ({ projectId, isSidebarOpen, setSidebarOpen, children }) => {
  return (
    <div
      css={{
        position: 'fixed',
        top: 0,
        left: 0,
        right: 0,
        height: constants.navbar.height,
        display: 'flex',
        justifyContent: 'space-between',
        alignItems: 'center',
        backgroundColor: colors.structure.mattGrey,
        boxShadow: constants.boxShadows.navBar,
        zIndex: zIndex.layout.navbar,
        paddingLeft: spacing.normal,
        paddingRight: spacing.normal,
      }}
    >
      <div css={{ display: 'flex', alignItems: 'stretch' }}>
        {!!projectId && (
          <button
            aria-label="Open Sidebar Menu"
            type="button"
            onClick={() => setSidebarOpen(!isSidebarOpen)}
            css={{
              border: 0,
              backgroundColor: 'inherit',
              color: colors.structure.steel,
              display: 'flex',
              justifyContent: 'center',
              alignItems: 'center',
              padding: spacing.base,
              marginLeft: -spacing.base,
              cursor: 'pointer',
            }}
          >
            <HamburgerSvg height={constants.icons.regular} />
          </button>
        )}

        <Link href="/" passHref>
          <a
            css={{
              paddingLeft: spacing.base,
              paddingRight: spacing.base,
              paddingBottom: spacing.mini,
              display: 'flex',
              alignItems: 'center',
            }}
            aria-label="Home"
          >
            <LogoSvg height={LOGO_HEIGHT} />
          </a>
        </Link>

        <ProjectSwitcher projectId={projectId} />
      </div>

      {children}
    </div>
  )
}

Navbar.propTypes = {
  projectId: PropTypes.string.isRequired,
  isSidebarOpen: PropTypes.bool.isRequired,
  setSidebarOpen: PropTypes.func.isRequired,
  children: PropTypes.node.isRequired,
}

export default Navbar
