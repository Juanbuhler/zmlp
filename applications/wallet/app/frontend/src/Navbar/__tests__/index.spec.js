import TestRenderer from 'react-test-renderer'

import Navbar from '..'

describe('<Navbar />', () => {
  it('should render properly', () => {
    const mockSetSelectedProject = jest.fn()
    const mockSetSidebarOpen = jest.fn()
    const mockLogout = jest.fn()

    const component = TestRenderer.create(
      <Navbar
        projects={[{ id: '1', name: 'project-name', selected: true }]}
        setSelectedProject={mockSetSelectedProject}
        isSidebarOpen
        setSidebarOpen={mockSetSidebarOpen}
        logout={mockLogout}
      />,
    )

    expect(component.toJSON()).toMatchSnapshot()
  })
})
