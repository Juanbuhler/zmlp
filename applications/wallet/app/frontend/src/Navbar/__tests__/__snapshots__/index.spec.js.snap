// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navbar /> should render properly 1`] = `
.emotion-5 {
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  height: 44px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #2e2e2e;
  box-shadow: 0 0 4px 0 rgba(0,0,0,0.5);
  z-index: 100;
  padding-left: 16px;
  padding-right: 16px;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
}

.emotion-0 {
  border: 0;
  background-color: inherit;
  color: #808080;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 8px;
  margin: 0;
  margin-left: -8px;
  cursor: pointer;
}

.emotion-1 {
  padding-left: 8px;
  padding-right: 8px;
}

.emotion-4 {
  position: relative;
}

.emotion-3 {
  border: 0;
  margin: 0;
  padding: 0;
  width: 28px;
  height: 28px;
  border-radius: 28px;
  color: #ffffff;
  background-color: #4d4948;
  font-weight: 700;
}

.emotion-3:hover {
  cursor: pointer;
}

<div
  className="emotion-5"
>
  <div
    className="emotion-2"
  >
    <button
      aria-label="Open Sidebar Menu"
      className="emotion-0"
      onClick={[Function]}
      type="button"
    >
      <svg
        aria-hidden={true}
        data-file-name="hamburger.svg"
        width={20}
      />
    </button>
    <div
      className="emotion-1"
    >
      <svg
        data-file-name="logo.svg"
        width={110}
      />
    </div>
  </div>
  <div
    className="emotion-4"
  >
    <button
      className="emotion-3"
      onBlur={[Function]}
      onClick={[Function]}
      type="button"
    >
      JD
    </button>
  </div>
</div>
`;
